tools/travis-before-install.sh

```diff
@@ -32,6 +36,16 @@ python -V
 
 popd
 
-pip install --upgrade pip setuptools
-pip install -r test_requirements.txt
+pip install --upgrade pip
+
+# 'setuptools', 'wheel' and 'cython' are build dependencies.  This information
+# is stored in pyproject.toml, but there is not yet a standard way to install
+# those dependencies with, say, a pip command, so we'll just hard-code their
+# installation here.  We only need to install them separately for the cases
+# where numpy is installed with setup.py. When pip is used, pip gets the build
+# dependencies from pyproject.toml.
+# We get the argument for 'cython' from test_requirements.txt to avoid
+# duplicating the version requirement stored there.
+pip install setuptools wheel `grep cython test_requirements.txt`
+
```

Comment:
Which run fails without this line?

Reply:
One or more of the `setup.py` installs... I think.  Let's find out...

Reply:
Actually, all the non-pip installs should fail, because travis-test.sh doesn't explicitly install cython.  The pip installs should work, because pip reads pyproject.toml and does a "private" install of the build requires that it find there.

Reply:
The two jobs that set `USE_WHEEL=1` and the one that sets `USE_DEBUG=1` fail to build numpy, because they build numpy by running `setup.py`, but cython is not installed.

Reply:
Maybe add that to the comment: `version requirement stored there.` -> `version requirement for the USE_WHEEL=1 and USE_DEBUG=1 builds.`

Reply:
Yeah, I can change that comment to be clearer.

The *version requirement* is for any build of numpy, not just those that set `USE_WHEEL` or `USE_DEBUG`.  I put that comment there as an attempt to explain why I get the cython package argument from `test_requirements.txt` instead of just writing

    pip install setuptools wheel cython

I'll work on better wording.

Even better would be to read the build requirements directly from pyproject.toml.  I had that in an earlier version of this PR, but I got the sense that it seemed too complicated.

(Unfortunately, we currently have the required version of cython hard-coded in three places: pyproject.toml, test_requirements.txt, and tools/cythonize.py.  Not sure it is worth the effort at the moment to fix that; even if it is, it can probably wait for a different PR.)

Reply:
I updated the comments.  Tests are running again...